cucumber-selenium-tests/src/test/java/com/skillstorm/PageObjects/BudgetsPage.java

```diff
@@ -0,0 +1,324 @@
+    private Navbar navbar;
+
+    // Edit and Delete Budget Modal Elements
+    private WebElement editBudgetButton;
+    private WebElement deleteBudgetButton;
+    private WebElement editedBudgetedInput;
+    private WebElement saveBudgetButton;
+    private WebElement confirmDeleteButton;
+    // Edit and Delete Savings Bucket Modal Elements
+    private WebElement editSavingsBucketButton;
+    private WebElement deleteSavingsBucketButton;
+    private WebElement editedAmountRequiredInput;
+    private WebElement saveSavingsBucketButton;
+    private WebElement confirmDeleteSavingsBucketButton;
+
+    private Map<String, WebElement> elementMap;
+    private List<WebElement> buttonList = new ArrayList<>();
+
+    public BudgetsPage(WebDriver driver) {
+        super(driver);
+        PageFactory.initElements(driver, this);
+        elementMap = new HashMap<>();
+        elementMap.put("summaryComponentDiv", summaryComponentDiv);
+        elementMap.put("addNewBudgetButton", addNewBudgetButton);
+        elementMap.put("budgetedInput", budgetedInput);
+        elementMap.put("budgetsTable", budgetsTable);
+        elementMap.put("savingsBucketTable", savingsBucketTable);
+        elementMap.put("budgetNameInput", budgetNameInput);
+        elementMap.put("amountRequiredInput", amountRequiredInput);
+        elementMap.put("submitNewSavingsBucketButton", submitNewSavingsBucketButton);
+        elementMap.put("editBudgetButton", editBudgetButton);
+        elementMap.put("deleteBudgetButton", deleteBudgetButton);
+        elementMap.put("editedBudgetedInput", editedBudgetedInput);
+        elementMap.put("saveBudgetButton", saveBudgetButton);
+        elementMap.put("confirmDeleteButton", confirmDeleteButton);
+        elementMap.put("editSavingsBucketButton", editSavingsBucketButton);
+        elementMap.put("deleteSavingsBucketButton", deleteSavingsBucketButton);
+        elementMap.put("editedAmountRequiredInput", editedAmountRequiredInput);
+        elementMap.put("saveSavingsBucketButton", saveSavingsBucketButton);
+        elementMap.put("confirmDeleteSavingsBucketButton", confirmDeleteSavingsBucketButton);
+        elementMap.put("editSpendingBudgetButton", editSpendingBudgetButton);
+        elementMap.put("totalBudgetAmountInput", totalBudgetAmountInput);
+        elementMap.put("submitSpendingBudgetButton", submitSpendingBudgetButton);
+        elementMap.put("spendingBudgetDiv", spendingBudgetDiv);
+    }
+
+    @Override
+    public List<Component> getChildComponents() {
+        ArrayList<Component> components = new ArrayList<Component>();
+        components.add(navbar);
+        return new ArrayList<Component>(Arrays.asList(navbar));
+    }
+
+    @Override
+    public Component getChildComponent(String name) {
+        if(name.equals("navbar")) return navbar;
+        return null;
+    }
+
+    @Override
+    public List<WebElement> getWebElements() {
```

Comment:
Medium impact: This method does not return a list of all the WebElements available in the BudgetsPage.
Suggestion: Get the collection of values from the elementsMap and convert the collection to an ArrayList (I believe ArrayList has a constructor that accepts a Collection as an argument).